### PR TITLE
flux: update to latest releases

### PR DESCRIPTION
**Pull request by SteVwonder**

### CODE REVIEW

File: var/spack/repos/builtin/packages/flux-core/package.py [hunk 1]
```diff
@@ -15,20 +15,15 @@ class FluxCore(AutotoolsPackage):
...
     version('0.17.0', sha256='3f8c6cb72982028f86a96c0098cacd3a6e9de359fa1cf077380c835a20e7b7f7')
     version('0.16.0', sha256='1582f7fb4d2313127418c34de7c9ce4f5fef00622d19cedca7bed929f4709f10')
     version('0.15.0', sha256='51bc2eae69501f802459fc82f191eb5e8ae0b4f7e9e77ac18543a850cc8445f5')
-    version('0.11.3', sha256='91b5d7dca8fc28a77777c4e4cb8717fc3dc2c174e70611740689a71901c6de7e')
-    version('0.11.2', sha256='ab8637428cd9b74b2dff4842d10e0fc4acc8213c4e51f31d32a4cbfbdf730412')
-    version('0.11.1', sha256='3c8495db0f3b701f6dfe3e2a75aed794fc561e9f28284e8c02ac67693bfe890e')
-    version('0.11.0', sha256='a4d8ff92e79b4ca19d556395bb8c5f8dc02fd9d5a8cc38c4a2c66867a96de5ea')
-    version('0.10.0', sha256='a70cdd228077af60c9443a5c69d3da932e447dd11697f5fef9028c48dabb3041')
-    version('0.9.0',  sha256='7b5b4aa72704b3c4432136b9e515e0d663568e6dbfc3ecd2f91c83b65841104e')
-    version('0.8.0',  sha256='eb4b0fe0da191acd3823ef42d415c40aae6a0c3aef62ebb27905658d045e11cc')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/flux-core/package.py:31]: Why are we removing these versions?

**Review by alalazo — CHANGES_REQUESTED** [review]
Can we avoid removing versions? If not, can you sign up to be a maintainer of the package and remove `flux-core@:0.11.3` mentions in other packages?

File: var/spack/repos/builtin/packages/flux-core/package.py (hunk 1 above)
Comment (SteVwonder) [reply var/spack/repos/builtin/packages/flux-core/package.py:31]: The versions removed are quite old.  We (the flux team) don't support them any more, and they have unique dependency requirements that complicate the Spack package. We don't know of any users that still use those versions.  If they didn't have significantly different dependencies, I wouldn't mind them being in the package.  Just trying to keep the flux-* packages more maintainable.  If removing versions is too onerous, I can back that change out.

> can you sign up to be a maintainer of the package

Sure! Happy to sign up for that.

>  remove flux-core@:0.11.3 mentions in other packages?

Good catch.  Sure, I will remove them, assuming you are OK with the version removal from flux-core.

Comment (alalazo) [reply var/spack/repos/builtin/packages/flux-core/package.py:31]: @SteVwonder If you sign up to be a maintainer of the package I'll redirect requests to restore old versions (if they ever happen) to you, so removing them now works for me 🙂

**Review by alalazo — APPROVED** [review]

**Comment by tldahlgren** [comment]
Thank you for your contribution.

**Pull request merged**